### ISSUE TITLE
haproxy uses 100% CPU

### ISSUE THREAD

**Issue by skliarie** [haproxy/haproxy#2402]
### Detailed Description of the Problem

I am using this helm to run haproxy: oci://registry-1.docker.io/bitnamicharts/haproxy
We have 80+ containers, but only one is having this problem, suggesting some specific request triggers the bug.

All containers have k8 ingress that terminates ssl, passing http to haproxy, thus excluding SSL-related issues.

Symptoms of the problem that shorly after start, haproxy uses unproportionally high amount of cpu, after some point reaching 100% CPU (or rather 400% as each pod is limited to 4 cores). It continues to serve requests for several hours, until (apparently) all threads are locked up, at this point requiring pod restart.

That haproxy receives handful of requests per hour, excluding overload-related issues.

I vaguely
I have no name!@haproxy-reports-nqa-6bd94cbf54-v2kxp:/$ cat /proc/sys/fs/file-max 
2015001
I have no name!@haproxy-reports-nqa-6bd94cbf54-v2kxp:/$ cat /proc/sys/fs/nr_open 
1048576
I have no name!@haproxy-reports-nqa-6bd94cbf54-v2kxp:/$ cat /proc/sys/fs/inotify/max_user_instances
8192

I am not sure, could be my imaginations, but it seems that last time it got stuck, I saw some message about no more free file handles or something like this. It is not visible each time.. Will try to catch that next time it stucks.

### Expected Behavior

cpu low and stable

### Steps to Reproduce the Behavior

1. start the pod
2. send requests and await 100% cpu
3. await it to stuck completely

### Do you have an idea how to solve the issue?

some highly specific request? Or not freeing some resource, eventually starving all threads?

### What is your configuration?

```haproxy
kubernetes v1.25.4
helm 3, using oci://registry-1.docker.io/bitnamicharts/haproxy

# helm list
NAME                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION             

haproxy-reports-nqa     default         1               2023-12-17 11:56:32.98813617 +0000 UTC  deployed        haproxy-0.10.10 2.9.1                   

values.yaml for the helm chart:
ingress:
  enabled: true
  annotations:
  ingressClassName: public
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt-production-ci
    nginx.ingress.kubernetes.io/ssl-redirect: "SSL_REDIRECT"
    # Always disable Strict-Transport-Security as legacy php 7.4 used on legacy farms does not know to deal with the header properly
    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
    nginx.ingress.kubernetes.io/configuration-snippet: |
      if ($scheme = https) {
        add_header  Strict-Transport-Security "max-age=0;";
      }
      more_clear_headers 'Strict-Transport-Security';
  hostname: haproxy-reports-nqa
  tls: true
service:
  type: ClusterIP
containerPorts:
  - name: http
    containerPort: 8080
configuration: |
  global
    log stdout format raw local0
    maxconn 1024
  defaults
    log global
    timeout client 60s
    timeout connect 60s
    timeout server 60s
  frontend fe_main
    bind :8080
    default_backend be_main
  backend be_main
    #server web1 10.3.1.95:80 check
    #server web2 10.3.1.96:80 check
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.9.1-f72603c 2023/12/15 - https://haproxy.org/
Status: stable branch - will stop receiving fixes around Q1 2025.
Known bugs: http://www.haproxy.org/bugs/bugs-2.9.1.html
Running on: Linux 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -Wfatal-errors -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference -fwrapv 
-Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wno-string-plus-int -Wno-at
omic-alignment
  OPTIONS = USE_THREAD=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_LUA=1 USE_PROMEX=1
  DEBUG   = -DDEBUG_STRICT -DDEBUG_MEMORY_POOLS

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE
 +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_AWSLC -OPENSSL_WOLFSSL -OT -PCRE -PCRE2 -PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL +PROMEX -P
THREAD_EMULATION -QUIC -QUIC_OPENSSL_COMPAT +RT +SHM_OPEN +SLZ +SSL -STATIC_PCRE -STATIC_PCRE2 -SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL -ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=16).
Built with OpenSSL version : OpenSSL 1.1.1w  11 Sep 2023
Running on OpenSSL version : OpenSSL 1.1.1w  11 Sep 2023
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.4.2
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built without PCRE or PCRE2 support (using libc's regex instead)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [BWLIM] bwlim-in
        [BWLIM] bwlim-out
        [CACHE] cache
        [COMP] compression
        [FCGI] fcgi-app
        [SPOE] spoe
        [TRACE] trace
```

### Last Outputs and Backtraces

```plain
# kubectl logs haproxy-reports-nqa-6bd94cbf54-v2kxp --tail 10 -f
Connect from 10.233.84.206:45918 to 10.233.86.53:8080 (fe_main/TCP)
Connect from 10.233.84.206:45536 to 10.233.86.53:8080 (fe_main/TCP)
Connect from 10.233.84.206:42076 to 10.233.86.53:8080 (fe_main/TCP)
Connect from 10.233.84.206:42084 to 10.233.86.53:8080 (fe_main/TCP)
```

**Comment by capflam** [comment]
Looking at your config, it seems your are using raw TCP frontend and backend.  Thus it may be the same issue than #2395. To be sure you may try to compile the HEAD of 2.9 or just to apply  5929895b4bac7f097a2ab886897f8068234fdadf on top of the 2.9.1. Another option is to disable the zero-copy forwarding by adding `-dZ` parameter on the command line or by adding `tune.disable-zero-copy-forwarding` in the global section. I

**Comment by brenc** [comment]
We ran into this too with 2.9.1. Had to kill haproxy completely to get rid of the 2.9.1 process which was stuck at 100% (even on our almost zero utilization staging instance). Example config in #2189. Will try HEAD of 2.9.

**Comment by Mickael-Martin** [comment]
Hello,
We got the same problem here. If you want some information, I can make tests.
"`tune.disable-zero-copy-forwarding` in the global section." works for me.

**Comment by agowa** [comment]
Same issue. Here are some perf stats. I had it enabled for a few seconds and the server was getting almost no traffic in that time (maybe one or two connections) and it's still utilizing 6 CPU cores to 100% each...

```
show info
Name: HAProxy
Version: 2.9.1-f72603c
Release_date: 2023/12/15
Nbthread: 6
Nbproc: 1
Process_num: 1
Pid: 8
Uptime: 0d 12h45m21s
Uptime_sec: 45921
Memmax_MB: 0
PoolAlloc_MB: 51
PoolUsed_MB: 51
PoolFailed: 0
Ulimit-n: 1048576
Maxsock: 1048576
Maxconn: 524262
Hard_maxconn: 524262
CurrConns: 2050
CumConns: 140464
CumReq: 3644
MaxSslConns: 0
CurrSslConns: 0
CumSslConns: 1350
Maxpipes: 0
PipesUsed: 0
PipesFree: 0
ConnRate: 0
ConnRateLimit: 0
MaxConnRate: 12
SessRate: 0
SessRateLimit: 0
MaxSessRate: 12
SslRate: 0
SslRateLimit: 0
MaxSslRate: 11
SslFrontendKeyRate: 0
SslFrontendMaxKeyRate: 6
SslFrontendSessionReuse_pct: 0
SslBackendKeyRate: 0
SslBackendMaxKeyRate: 0
SslCacheLookups: 145
SslCacheMisses: 2
CompressBpsIn: 0
CompressBpsOut: 0
CompressBpsRateLim: 0
Tasks: 2088
Run_queue: 2048
Idle_pct: 3
node: 24915d19d256
Stopping: 0
Jobs: 2060
Unstoppable Jobs: 1
Listeners: 8
ActivePeers: 0
ConnectedPeers: 0
DroppedLogs: 1
BusyPolling: 0
FailedResolutions: 0
TotalBytesOut: 67215303
TotalSplicedBytesOut: 0
BytesOutRate: 0
DebugCommandsIssued: 0
CumRecvLogs: 0
Build info: 2.9.1-f72603c
Memmax_bytes: 0
PoolAlloc_bytes: 53742560
PoolUsed_bytes: 53742560
Start_time_sec: 1704730493
Tainted: 0
TotalWarnings: 3
MaxconnReached: 0
BootTime_ms: 15522
Niced_tasks: 0

show pools
Dumping pools usage. Use SIGQUIT to flush them.
  - Pool hlua_queuew (24 bytes) : 26 allocated (624 bytes), 26 used (~25 by thread caches), needed_avg 2, 0 failures, 7 users, @0x7fa0ff23c200 [SHARED]
  - Pool mux_pt (32 bytes) : 4116 allocated (131712 bytes), 4116 used (~8 by thread caches), needed_avg 3988, 0 failures, 3 users, @0x7fa0ff1e3280 [SHARED]
  - Pool comp_state (40 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 2 users, @0x7fa0ff1de980 [SHARED]
  - Pool hlua_flt_ct (48 bytes) : 4119 allocated (197712 bytes), 4119 used (~15 by thread caches), needed_avg 3991, 0 failures, 8 users, @0x7fa0ff241200 [SHARED]
  - Pool conn_hash_n (56 bytes) : 10313 allocated (577528 bytes), 10313 used (~49 by thread caches), needed_avg 10185, 0 failures, 3 users, @0x7fa0ff1ff280 [SHARED]
  - Pool hlua_esub (64 bytes) : 30 allocated (1920 bytes), 30 used (~29 by thread caches), needed_avg 6, 0 failures, 4 users, @0x7fa0ff24b1c0 [SHARED]
  - Pool fcgi_hdr_ru (72 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 2 users, @0x7fa0ff1da080 [SHARED]
  - Pool stconn (80 bytes) : 4118 allocated (329440 bytes), 4118 used (~14 by thread caches), needed_avg 3990, 0 failures, 3 users, @0x7fa0ff1ec980 [SHARED]
  - Pool sedesc (88 bytes) : 4118 allocated (362384 bytes), 4118 used (~14 by thread caches), needed_avg 3990, 0 failures, 1 users, @0x7fa0ff1e8080 [SHARED]
  - Pool comp_ctx (96 bytes) : 1 allocated (96 bytes), 1 used (~0 by thread caches), needed_avg 0, 0 failures, 2 users, @0x7fa0ff1d0980 [SHARED]
  - Pool ehdl_sub (104 bytes) : 1 allocated (104 bytes), 1 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff1ac080 [SHARED]
  - Pool ssl_sock_ct (120 bytes) : 11 allocated (1320 bytes), 11 used (~10 by thread caches), needed_avg 1, 0 failures, 1 users, @0x7fa0ff2501c0 [SHARED]
  - Pool uniqueid (128 bytes) : 8286 allocated (1060608 bytes), 8286 used (~41 by thread caches), needed_avg 8158, 0 failures, 3 users, @0x7fa0ff2292c0 [SHARED]
  - Pool hlua (144 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff246200 [SHARED]
  - Pool pp_tlv_128 (152 bytes) : 2063 allocated (313576 bytes), 2063 used (~11 by thread caches), needed_avg 1935, 0 failures, 2 users, @0x7fa0ff1fa980 [SHARED]
  - Pool h2s (168 bytes) : 29 allocated (4872 bytes), 29 used (~29 by thread caches), needed_avg 4, 0 failures, 1 users, @0x7fa0ff22e080 [SHARED]
  - Pool fcgi_strm (184 bytes) : 70 allocated (12880 bytes), 70 used (~66 by thread caches), needed_avg 17, 0 failures, 3 users, @0x7fa0ff2372c0 [SHARED]
  - Pool spoe_appctx (200 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff21b240 [SHARED]
  - Pool tcpcheck_ru (208 bytes) : 4124 allocated (857792 bytes), 4124 used (~21 by thread caches), needed_avg 3996, 0 failures, 3 users, @0x7fa0ff2249c0 [SHARED]
  - Pool http_txn (216 bytes) : 32 allocated (6912 bytes), 32 used (~31 by thread caches), needed_avg 6, 0 failures, 1 users, @0x7fa0ff216940 [SHARED]
  - Pool ssl_sock_cl (256 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff2551c0 [SHARED]
  - Pool spoe_ctx (264 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff204040 [SHARED]
  - Pool appctx (272 bytes) : 14 allocated (3808 bytes), 14 used (~13 by thread caches), needed_avg 1, 0 failures, 1 users, @0x7fa0ff1cc080 [SHARED]
  - Pool pp_tlv_256 (280 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff1f6040 [SHARED]
  - Pool resolv_reso (448 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff208940 [SHARED]
  - Pool resolv_answ (600 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff20d280 [SHARED]
  - Pool ehdl_sub_ed (784 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff1d52c0 [SHARED]
  - Pool stream (880 bytes) : 2056 allocated (1809280 bytes), 2056 used (~4 by thread caches), needed_avg 1928, 0 failures, 1 users, @0x7fa0ff212040 [SHARED]
  - Pool requri (1024 bytes) : 32 allocated (32768 bytes), 32 used (~31 by thread caches), needed_avg 4, 0 failures, 1 users, @0x7fa0ff1c2dc0 [SHARED]
  - Pool fcgi_conn (1216 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff220080 [SHARED]
  - Pool h2c (1288 bytes) : 7 allocated (9016 bytes), 7 used (~7 by thread caches), needed_avg 1, 0 failures, 1 users, @0x7fa0ff232980 [SHARED]
  - Pool hpack_tbl (4096 bytes) : 7 allocated (28672 bytes), 7 used (~7 by thread caches), needed_avg 1, 0 failures, 1 users, @0x7fa0ec6c6080 [SHARED]
  - Pool dns_msg_buf (65544 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff1f1280 [SHARED]
  - Pool buffer (16000000 bytes) : 2 allocated (32000000 bytes), 2 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff1c76c0 [SHARED]
  - Pool trash (16000032 bytes) : 1 allocated (16000032 bytes), 1 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7fa0ff1b09c0
Total: 35 pools, 53743056 bytes allocated, 53743056 used (~121216 by thread caches).

show activity
thread_id: 3 (1..6)
date_now: 1704776539.078827
uptime_now: 46029.859775
ctxsw: 2174743684 [ 1526471770 3390478410 415650453 298678265 2201633506 2931765872 ]
tasksw: 1192818 [ 153079 153154 198770 152895 290215 244705 ]
empty_rq: 824 [ 92 87 86 163 185 211 ]
long_rq: 1922608821 [ 342646142 360907386 331898345 294811627 270643609 321701712 ]
curr_rq: 2048 [ 340 341 343 341 342 341 ]
loops: 1922622593 [ 342646283 360907513 331899699 294811888 270652007 321705203 ]
wake_tasks: 1922608821 [ 342646142 360907386 331898345 294811627 270643609 321701712 ]
wake_signal: 0 [ 0 0 0 0 0 0 ]
poll_io: 134948 [ 10535 10552 33354 10684 36460 33363 ]
poll_exp: 0 [ 0 0 0 0 0 0 ]
poll_drop_fd: 1 [ 0 0 0 1 0 0 ]
poll_skip_fd: 0 [ 0 0 0 0 0 0 ]
conn_dead: 0 [ 0 0 0 0 0 0 ]
stream_calls: 912168 [ 152014 152108 152019 151891 152003 152133 ]
pool_fail: 0 [ 0 0 0 0 0 0 ]
buf_wait: 0 [ 0 0 0 0 0 0 ]
cpust_ms_tot: 40592981 [ 6851233 6773047 6725064 6790922 6719237 6733478 ]
cpust_ms_1s: 1397 [ 20 347 48 413 272 297 ]
cpust_ms_15s: 14723 [ 1781 2808 1897 2923 3117 2197 ]
avg_cpu_pct: 90 [ 99 85 99 86 99 69 ]
avg_loop_us: 232 [ 218 179 218 337 166 271 ]
accepted: 3607 [ 582 633 538 580 630 644 ]
accq_pushed: 3607 [ 605 602 607 598 597 598 ]
accq_full: 0 [ 0 0 0 0 0 0 ]
accq_ring: 0 [ 0 0 0 0 0 0 ]
fd_takeover: 55 [ 17 10 5 7 7 9 ]
check_adopted: 0 [ 0 0 0 0 0 0 ]
check_started: 137184 [ 0 0 22868 0 68588 45728 ]
check_active: 0 [ 0 0 0 0 0 0 ]
check_running: 0 [ 0 0 0 0 0 0 ]

set profiling tasks on

set profiling tasks off

show profiling tasks
Tasks activity over 60.093 sec till 6.725 sec ago:
  function                      calls   cpu_tot   cpu_avg   lat_tot   lat_avg
  sc_conn_io_cb              56088763   2.627m    2.810us   17.09h    1.097ms <- sc_app_chk_rcv_conn@src/stconn.c:783 tasklet_wakeup
  sc_conn_io_cb              56088717   1.944m    2.079us   17.08h    1.096ms <- se_nego_ff@include/haproxy/stconn.h:520 tasklet_wakeup
  process_stream                 1276   1.244s    975.2us   529.0ms   414.6us <- sc_notify@src/stconn.c:1152 task_queue
  process_stream                 1193   1.255s    1.052ms   401.4ms   336.5us <- run_tasks_from_lists@src/task.c:657 task_queue
  process_chk                     357   152.6ms   427.5us   112.4ms   314.8us <- wake_srv_chk@src/check.c:1089 task_wakeup
  process_stream                    8   2.186ms   273.3us   1.004ms   125.6us <- sc_notify@src/stconn.c:1137 task_wakeup
  h2_io_cb                          7   937.4us   133.9us   16.62ms   2.374ms <- h2c_restart_reading@src/mux_h2.c:743 tasklet_wakeup
  ssl_sock_io_cb                    7   1.424ms   203.5us   222.5us   31.79us <- sock_conn_iocb@src/sock.c:877 tasklet_wakeup
  srv_cleanup_idle_conns            5   14.60us   2.919us   529.3us   105.9us <- run_tasks_from_lists@src/task.c:657 task_queue
  h2_io_cb                          5   7.483ms   1.497ms   2.142ms   428.4us <- ssl_sock_io_cb@src/ssl_sock.c:6613 tasklet_wakeup
  h1_timeout_task                   4   3.458us   864.0ns   2.132ms   533.0us <- h1_release@src/mux_h1.c:1056 task_wakeup
  sc_conn_io_cb                     3   351.1us   117.0us   1.247ms   415.6us <- h1_wake_stream_for_recv@src/mux_h1.c:3188 tasklet_wakeup
  accept_queue_process              3   380.1us   126.7us   189.0us   63.00us <- listener_accept@src/listener.c:1476 tasklet_wakeup
  h1_io_cb                          3   186.8us   62.27us   2.944us   981.0ns <- sock_conn_iocb@src/sock.c:877 tasklet_wakeup
  srv_cleanup_toremove_conns        3   1.073ms   357.6us   2.671ms   890.3us <- srv_cleanup_idle_conns@src/server.c:6615 task_wakeup
  process_stream                    3   12.68ms   4.227ms   222.1us   74.02us <- stream_new@src/stream.c:578 task_wakeup
  h1_timeout_task                   2   752.8us   376.4us   4.547ms   2.274ms <- h1_init@src/mux_h1.c:989 task_queue
  ssl_sock_io_cb                    2   32.27ms   16.14ms   709.6us   354.8us <- ssl_sock_start@src/ssl_sock.c:5864 tasklet_wakeup
  h2_timeout_task                   2   9.357us   4.678us   1.138ms   568.8us <- h2_release@src/mux_h2.c:1154 task_wakeup
  sc_conn_io_cb                     2   123.6us   61.80us   703.1us   351.6us <- sock_conn_iocb@src/sock.c:877 tasklet_wakeup
  h2_io_cb                          2   1.373us   686.0ns   1.469ms   734.6us <- h2_snd_buf@src/mux_h2.c:6903 tasklet_wakeup
  srv_cleanup_idle_conns            2   19.77us   9.883us   895.5us   447.8us <- srv_add_to_idle_list@src/server.c:6525 task_schedule
  sc_conn_io_cb                     2   206.6us   103.3us   81.34us   40.67us <- h1_wake_stream_for_send@src/mux_h1.c:3198 tasklet_wakeup
  h1_io_cb                          2   83.20us   41.60us   1.904us   952.0ns <- sock_conn_iocb@src/sock.c:856 tasklet_wakeup
  task_run_applet                   1   962.0ns   962.0ns   74.52us   74.52us <- sc_app_shut_applet@src/stconn.c:914 appctx_wakeup
  task_run_applet                   1   40.60us   40.60us   97.45us   97.45us <- sc_applet_create@src/stconn.c:512 appctx_wakeup
  task_run_applet                   1   154.1us   154.1us   622.1us   622.1us <- sc_app_chk_snd_applet@src/stconn.c:975 appctx_wakeup
```

And when running `perf top -a -F max --hierarchy` I get this at the top:
```
-   77.40%     haproxy                                                                                                                                                                        ◆
        7.30%     [.] 0x00000000001c839f                                                                                                                                                      
        5.41%     [.] 0x000000000021c72b                                                                                                                                                      
        5.06%     [.] 0x000000000021c716                                                                                                                                                      
```

And I'm using this docker image:
```
haproxy            alpine       b6644dc14260   2 weeks ago    25.8MB
```

It looks like adding `tune.disable-zero-copy-forwarding` to global works - for now at least. Hard to tell yet, as just restarting haproxy alone without adding that change to the config was also enough to get it to behave for a while.

**Comment by capflam** [comment]
Thanks for report. This issue is definitely a duplicate of #2395. This bug was fixed and the fix will be shipped with the 2.9.2 planned for this week.
